### ISSUE TITLE
tf_upgrade_v2 has no attribute __pasta__

### ISSUE THREAD

**Issue by kriskorrel-cw** [tensorflow/tensorflow#29070]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.14.5
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-alpha0
- Python version: 3.7.3
- GPU model and memory: cpu

**Describe the current behavior**
When running the tf2 upgrade script on any of my projects I would get the following error:
```
$ tf_upgrade_v2 --intree my_project --outtree my_project_upgrade

INFO line 46:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 49:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 30:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 33:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 21:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 24:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 21:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 24:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 35:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 38:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 68:17: Added keywords to args of function 'tf.convert_to_tensor'
INFO line 69:17: Added keywords to args of function 'tf.convert_to_tensor'
INFO line 72:34: Renamed 'tf.log' to 'tf.math.log'
INFO line 75:21: Added keywords to args of function 'tf.reduce_max'
INFO line 76:15: Added keywords to args of function 'tf.reduce_mean'
INFO line 56:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 59:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 75:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 78:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 41:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 43:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 65:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 67:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 41:13: Renamed 'tf.ConfigProto' to 'tf.compat.v1.ConfigProto'
INFO line 43:11: Renamed 'tf.Session' to 'tf.compat.v1.Session'
INFO line 45:22: Renamed 'tf.Summary' to 'tf.compat.v1.Summary'
INFO line 84:22: Renamed 'tf.Summary' to 'tf.compat.v1.Summary'
Traceback (most recent call last):
  File "/usr/local/bin/tf_upgrade_v2", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/tf_upgrade_v2_main.py", line 123, in main
    args.input_tree, output_tree, args.copy_other_files)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 624, in process_tree
    _, l_report, l_errors = self.process_file(input_path, output_path)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 494, in process_file
    temp_file)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 548, in process_opened_file
    self.update_string_pasta("".join(lines), in_filename))
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 510, in update_string_pasta
    t = pasta.parse(text)
  File "/usr/local/lib/python3.7/site-packages/pasta/__init__.py", line 25, in parse
    annotator.visit(t)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 220, in visit_Module
    self.generic_visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 270, in generic_visit
    self.visit(item)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 95, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 381, in visit_ClassDef
    self.visit(stmt)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 95, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 411, in visit_FunctionDef
    self.visit(stmt)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 530, in visit_Assign
    self.visit(node.value)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 829, in visit_IfExp
    self.visit(node.body)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 700, in visit_BoolOp
    self.visit(value)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1190, in visit
    super(AstAnnotator, self).visit(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 132, in visit
    super(BaseVisitor, self).visit(node)
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ast.py", line 262, in visit
    return visitor(node)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 803, in visit_Dict
    self.visit(key)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/annotate.py", line 1188, in visit
    fmt.set(node, 'indent', self._indent)
  File "/usr/local/lib/python3.7/site-packages/pasta/base/formatting.py", line 37, in set
    _formatting_dict(node)[name] = value
  File "/usr/local/lib/python3.7/site-packages/pasta/base/formatting.py", line 49, in _formatting_dict
    return getattr(node, PASTA_DICT)
AttributeError: 'NoneType' object has no attribute '__pasta__'
```

I have tried updating pasta (`pip3 install --upgrade pasta`) which now results in 
```
$ tf_upgrade_v2 --intree my_project --outtree my_project_upgrade

Traceback (most recent call last):
  File "/usr/local/bin/tf_upgrade_v2", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/tf_upgrade_v2_main.py", line 123, in main
    args.input_tree, output_tree, args.copy_other_files)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 624, in process_tree
    _, l_report, l_errors = self.process_file(input_path, output_path)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 494, in process_file
    temp_file)
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 548, in process_opened_file
    self.update_string_pasta("".join(lines), in_filename))
  File "/usr/local/lib/python3.7/site-packages/tensorflow/tools/compatibility/ast_edits.py", line 510, in update_string_pasta
    t = pasta.parse(text)
AttributeError: module 'pasta' has no attribute 'parse'
```
I was unable to find anyone with similar problems.
What is the supported pasta version that is needed to run the tf2 upgrade script?

**Comment by achandraa** [comment]
tf_upgrade_v2 utility is included automatically with pip install of TF2.0. Can you please try re-installing TF version and let us know if that resolves the issue. You can also refer this [link](https://medium.com/tensorflow/upgrading-your-code-to-tensorflow-2-0-f72c3a4d83b5) to more information. Thanks!

**Comment by kriskorrel-cw** [comment]
No, not resolved. I believe it is linked to #26486

**Comment by achandraa** [comment]
@kriskorrel-cw : Just to verify whether it is fixed or not, can you try with pasta - 0.1.7 and let us know if that resolves it. Thanks!

**Comment by kriskorrel-cw** [comment]
This particular issue is resolved with pasta 0.1.7, yes. However, it does raise another issue for which I shall open a new ticket :)

**Comment by ivan866** [comment]
same issue. win10 x64; tf-gpu 2.2.0; pasta 0.1 / 0.2
